### ISSUE TITLE
[Code Style] Give users more control over the visual treatments of each option

### ISSUE THREAD

**Issue by dpoeschl** [dotnet/roslyn#13963]
On the Code Style page today, we allow users to choose their style choice and their severity.

![image](https://cloud.githubusercontent.com/assets/235241/18728865/14715dd8-8004-11e6-9233-b0532eeddd36.png)

It would be nice if they could also control the "visual treatment" (working title) of each. Examples include
1. Inclusion in the Error List
2. Presence of the "..." or squiggle indicators
3. Fading out of non-compliant code
4. Inclusion in the scroll bar

Another open question is whether this level of control should apply to only things on the Code Style page, or all analyzers.

**Comment by dpoeschl** [comment]
Based on conversation with @CyrusNajmabadi @Pilchie @DustinCampbell @kuhlenh

**Comment by CyrusNajmabadi** [comment]
Here's an idea:

The code style page currently has 2 columns.  One for preference and one for severity.  I could imagine a 3rd column.  This column would indicate "presentation" or "location" (or some good name).  Each style would have a dropdown with checkboxes in it allowing them to control the above options that david laid out (i.e. Show in error list, Show squiggles, etc.)

We would pick the best defaults for each style option we provided, but users could tweak as they want.  So, for example, "unused usings" would just have the "fade out" presentation option checked.  But users could say "i want this in the scroll bar and error list if they wanted."

Primary concern would be that this would bloat the size of this dialog.  However, my hope would be that by the time we did this, we could add the extra inch or so to make this possible.
